From: AMEET M. PARANJAPE <aparanja@redhat.com>
Date: Wed, 26 Nov 2008 21:47:10 -0500
Subject: [misc] fix add return signal to ptrace_report_exec
Message-id: 20081127024650.8916.70921.sendpatchset@squad5-lp1.lab.bos.redhat.com
O-Subject: [PATCH RHEL5.3 BZ471112] fix add return signal to ptrace_report_exec()
Bugzilla: 471112
RH-Acked-by: Jerome Marchand <jmarchan@redhat.com>
RH-Acked-by: Oleg Nesterov <oleg@redhat.com>
RH-Acked-by: David Howells <dhowells@redhat.com>
RH-Acked-by: Roland McGrath <roland@redhat.com>

RHBZ#:
======
https://bugzilla.redhat.com/show_bug.cgi?id=471112

Description:
===========
In function ptrace_report_exec() provide a return value after sending the
signal.

RHEL Version Found:
================
RHEL 5.3 snapshot 1

kABI Status:
============
No symbols were harmed.

Brew:
=====
Built on all platforms.
http://brewweb.devel.redhat.com/brew/taskinfo?taskID=1589740

Upstream Status:
================
N/A.  This fix is specific to the ptrace version shipped with RHEL5.x.

Test Status:
============
Start gbd on ppc system.  The gdb session would hang and then panic when
trying to kill the session with a kill -9.

With this patch applied the hang is not seen.  The crash is also not
reproducible (though this patch probably didn't fix that).
===============================================================
Ameet Paranjape 978-392-3903 ext 23903
IBM on-site partner

Proposed Patch:
===============

diff --git a/kernel/ptrace.c b/kernel/ptrace.c
index a874696..3001d81 100644
--- a/kernel/ptrace.c
+++ b/kernel/ptrace.c
@@ -1977,6 +1977,7 @@ ptrace_report_exec(struct utrace_attached_engine *engine,
 	 * what ptrace can do with tsk->exit_code there.
 	 */
 	send_sig(SIGTRAP, tsk, 0);
+	return UTRACE_ACTION_RESUME;
 }
 
 static u32
